File: 김하영/Barkingdog_Stack/6198.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";

const input = fs.readFileSync(filePath).toString().trim().split("\n");

const N = input.shift();
const inputArr = input.map((str) => Number(str.trim()));

const solution = (inputArr) => {
  let result = 0;
  const stack = [];

  for (let i = 0; i < inputArr.length; i++) {
    while (stack.length && stack[stack.length - 1] <= inputArr[i]) {
      stack.pop();
    }
    stack.push(inputArr[i]);
    console.log(stack.length - 1);
    result += stack.length - 1;
  }

  return result;
};

console.log(solution(inputArr));

File: BoSeonPark/BarkingDog_스택/6198.py
import sys
input = sys.stdin.readline

n = int(input())
answer = 0
stack = []
building = [int(input()) for _ in range(n)]

for i in range(n):
    while stack:
        if stack[-1] <= building[i]:
            stack.pop()
        else:
            break
    stack.append(building[i])
    answer += len(stack) - 1
print(answer)